File: THTR.AI/THTR/Models/Character.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel

class Character(BaseModel):
    id: UUID

    first_name: Optional[str] = None
    last_name: Optional[str] = None
    backstory: Optional[str] = None

    age: Optional[int] = None
    location: Optional[str] = None
    identity: Optional[str] = None
    education: Optional[str] = None
    occupation: Optional[str] = None
    family: Optional[str] = None
    personality: Optional[str] = None
    worldview: Optional[str] = None

    class Config:
        extra = "forbid"

File: THTR.Common/DTOs/Entities/Character.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace THTR.Common.DTOs.Entities
{
    public class Character
    {        
        public Guid id { get; set; }        
        public string? first_name { get; set; }
        public string? last_name { get; set; }
        public string? backstory { get; set; }
        public int? age { get; set; }
        public string? location { get; set; }
        public string? identity { get; set; }
        public string? education { get; set; }
        public string? occupation { get; set; }
        public string? family { get; set; }
        public string? personality { get; set; }
        public string? worldview { get; set; }
    }
}
